### PR TITLE
EDU-1515 Different approach to fixing PDF downscaling with the viewport

### PR DESCRIPTION
**Pull request by adrianaluchian**
Closes https://educandu.atlassian.net/browse/EDU-1515

### CODE REVIEW

File: src/components/pdf-document.js [hunk 1]
```diff
@@ -106,8 +116,8 @@ function PdfDocument({ file, pageNumber, stretchDirection, showTextOverlay, onLo
             <Page
               key={actualPageNumber}
               pageNumber={actualPageNumber}
-              height={stretchDirection === ORIENTATION.vertical ? containerHeight : null}
-              width={stretchDirection === ORIENTATION.horizontal ? containerWidth : null}
+              height={null}
+              width={containerWidth}
```
Comment (adrianaluchian) [inline src/components/pdf-document.js:120]: I went for simplifying the component by dropping this never used case. Hope that's alright for you.

Comment (ahelmberger) [reply src/components/pdf-document.js:120]: Oh, more than alright :-)

File: src/components/pdf-document.js [hunk 2]
```diff
@@ -38,14 +39,23 @@ function PdfDocument({ file, pageNumber, stretchDirection, showTextOverlay, onLo
   }, [pageNumber]);
 
   useEffect(() => {
-    const handleResize = () => setResizeRenderKey(oldKey => oldKey + 1);
+    const handleResize = () => {
+      const viewerBoundingClientRect = viewerRef.current.getBoundingClientRect();
+      const currentViewerViewportPadding = window.innerWidth - viewerBoundingClientRect.width;
+
+      const viewerWidthOverflow = viewerViewportPadding - currentViewerViewportPadding;
+      if (viewerWidthOverflow > 0) {
```
Comment (ahelmberger) [inline src/components/pdf-document.js:47]: Nicer and cleaner solution :+1:

**Review by ahelmberger — APPROVED** [review]

File: src/components/pdf-document.js (hunk 2 above)
Comment (adrianaluchian) [reply src/components/pdf-document.js:47]: 🙇

**Pull request merged**